samples/nrf9160/multicell_location/src/main.c

```diff
@@ -314,13 +332,18 @@ void main(void)
 
 	LOG_INF("Connected to LTE network");
 
-	if (IS_ENABLED(CONFIG_MULTICELL_LOCATION_SAMPLE_REQUEST_PERIODIC)) {
-		LOG_INF("Requesting neighbor cell information every %d seconds",
-			CONFIG_MULTICELL_LOCATION_SAMPLE_REQUEST_PERIODIC_INTERVAL);
-		k_work_schedule(&periodic_search_work, K_NO_WAIT);
-	} else {
-		start_cell_measurements();
-	}
+	/* Wait until RRC connection has been released, otherwise modem will not be able to
+	 * measure neighbor cells unless currently configured by the network.
+	 */
+	k_sem_take(&rrc_idle, K_FOREVER);
```

Comment:
That's the way I first tried to do it, but the problem is, that the sample calls `start_cell_measurements()` in the system workqueue, which is the same workqueue (=thread) used by LTE LC to send the event that gives the semaphore. So, the sample ends up in a deadlock waiting for the semaphore. The sample would have to have a separate workqueue to make it work.

When `start_cell_measurements()` is called from the workqueue, it happens either because of periodic search or a cell change. In both cases the RRC connection is normally inactive, so the current implementation should be good enough in most cases.

If/when the wait is moved into LTE LC, this would not be a problem anymore. The `lte_lc_neighbor_cell_measurement()` can return immediately and we can just delay the `LTE_LC_EVT_NEIGHBOR_CELL_MEAS` event.